### PR TITLE
MAP-813 java update

### PR DESCRIPTION
**Pull request by michalnawrockidj**

### CODE REVIEW

**Review by GurnankCheema — APPROVED** [review]
LGTM

File: build.gradle.kts [hunk 1]
```diff
@@ -44,33 +42,20 @@ dependencies {
   implementation("com.google.code.gson:gson")
   implementation("com.google.guava:guava:33.0.0-jre")
   implementation("org.jetbrains.kotlin:kotlin-reflect")
-  implementation("io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:1.32.0")
-  implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.7.3")
+  implementation("io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:2.1.0")
```
Comment (petergphillips) [inline build.gradle.kts:45]: this should really still be at `1.32` to match app insights

File: build.gradle.kts [hunk 2]
```diff
@@ -82,13 +67,16 @@ allOpen {
...
+  toolchain.languageVersion.set(JavaLanguageVersion.of(21))
 }
 
 tasks {
   withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
-    kotlinOptions {
-      jvmTarget = "19"
+    compilerOptions {
+      apiVersion.set(org.jetbrains.kotlin.gradle.dsl.KotlinVersion.KOTLIN_2_1)
+      freeCompilerArgs.set(listOf("-Xemit-jvm-type-annotations"))
```
Comment (petergphillips) [inline build.gradle.kts:77]: out of interest what did you need all of this for?  We normally just do https://github.com/ministryofjustice/hmpps-prisoner-location-api/blob/main/build.gradle.kts#L39-L49

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/utils/JwtAuthHelper.kt [hunk 3]
```diff
@@ -15,7 +14,7 @@ import java.util.UUID
 
 @Component
 class JwtAuthHelper {
-  private val keyPair: KeyPair
+  private final var keyPair: KeyPair
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/utils/JwtAuthHelper.kt:17]: we've also improved the code in https://github.com/ministryofjustice/hmpps-prisoner-location-api/blob/main/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerlocationapi/config/JwtAuthHelper.kt#L17 to get rid of the `init` and avoid the intellij warnings

**Review by petergphillips — DISMISSED** [review]
LGTM

File: build.gradle.kts (hunk 2 above)
Comment (michalnawrockidj) [reply build.gradle.kts:76]: fixed

File: build.gradle.kts (hunk 1 above)
Comment (michalnawrockidj) [reply build.gradle.kts]: I left 1.32 now

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/utils/JwtAuthHelper.kt (hunk 3 above)
Comment (michalnawrockidj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/utils/JwtAuthHelper.kt]: I used your example.

**Review by petergphillips — APPROVED** [review]
LGTM

**Review by GurnankCheema — COMMENTED** [review]
lgtm

**Pull request merged**